src/Access/AccessControl.cpp

```diff
@@ -396,6 +396,7 @@ void AccessControl::addDiskStorage(const String & storage_name_, const String &
         }
     }
     auto new_storage = std::make_shared<DiskAccessStorage>(storage_name_, directory_, *changes_notifier, readonly_, allow_backup_);
+    new_storage->reload(ReloadMode::ALL);
```

Comment:
It seems this `reload()` is not needed because `DiskAccessStorage` loads its data in its constructor.